src/extractor.hpp

```diff
@@ -39,10 +43,13 @@ struct Extractor final : osmium::handler::Handler
      *
      * @param way the current way being processed
      */
+    bool FilterWay(const osmium::Way &way);
     void way(const osmium::Way &way);
+    void ParseTags(std::ifstream &tagfile);
```

Comment:
These two new methods might be better off being private (or at least protected).  Not sure we'd ever want to call them from outside the class constructor, they're mostly just internal helpers.